src/macros.rs

```diff
@@ -0,0 +1,73 @@
+//! Convenience macros
+
+/// Macro to produce an array of [`VertexAttributeDescriptor`]
+///
+/// Output has type: `[VertexAttributeDescriptor; _]`. Usage is as follows:
+/// ```
+/// let attrs = vertex_attr_array![Float2, Float, Ushort4];
+///
+/// // In case the shader location does not start from 0, it may be specified:
```

Comment:
So the limitation here is that all of the shader attributes have to be continuous in terms of shader locations. This is ok for our examples, and likely OK for other applications, but it introduces the restrictions we shouldn't have otherwise.

Would you be open to refactor the macro so that the locations are always explicitly provided for each attribute?

Reply:
Sure can do. If you can suggest syntax, even better.

As I understand it, we are always counting from a specified start-point, so maybe something like `[2..; Float2, Float3]` would be clear enough?

Reply:
we aren't counting anything, we are just providing locations specified in the shaders, they can be anything

Reply:
Aha, so a format like `[2, Float2; 3, Ushort4]` is preferable?

Reply:
yes, it looks good to me

Reply:
I implemented a slightly different syntax, in part to be more "rustic" (Rust does not have map literals) and in part because `:` cannot follow an `expr` (which would restrict the index to being a literal).